pkg/services/signingkeys/signingkeysimpl/service_test.go

```diff
@@ -0,0 +1,233 @@
+H0G37lOiboxmMWpLI2XOrNAlTYmDCVVTSjoF0zxvMzIyvRV488X6tI8LAIf3QjDj
++6IrJH1RF1AGwLSeD07JWq4F3epg6BwEXlMePCwUr8cUYAIrPlGWT/ywP9ZKX5Wt
+mNEZEgaWAohvdXGbkG4cQuIT2fvd2HvYDjbr9CvQDV5tHIE36jUrlbzVRHYxp0QQ
+XbPTTN9On6fSueYoFy47CtXJOHrbZ+r74CU0yHl7AoIBACwQYl7YzerTlEiyhB/g
+niAnQ1ia5JfdbmRwNQ8dw1avHXkZrP3xjaVmNe5CU5qsfzseqm3i9iGH2uJ5uN1A
+R0Wc6lyHcbje2JQIEx090rl9T0kDcghusMQa7Hko438uo3TcxfbdL1XyxZR+JBD+
+A6adWnlSNx9oib9113pp3C1NlwJeH+Hi27r6cdiBoJYPilu6Q7AqnmAo55J27H4C
+VXoB+9j7at77Rmu6k6jLKdBHBvccRe/Fe2HnIy8ZLycgglHEcfp3SUWZLoXPABXf
+5mx8rOB21e/yJy6mhObBV77dz+XLdcXduSf51VwDm5fkKSaL8F0ZYvnS+dbTUSfV
+f7sCggEAOQPw/jRPARf++TlLpyngkDV6Seud0EOfk0Nu59a+uOPAfd5ha1yBHGsk
+wOr9tGXZhR3b3LwwKczQrm7X8UjE6MzU6M7Zf9DylORNPPSVrkzYgszYNwCxHxF/
+15rBVbcBhDc6CUeSZcxVas9hvOslGdu0HzrIcqSDw2hBwHR6hQvBfOcGr1ldAcvp
+BstdZBY6B3nuDhtNiUn544K7BaJlPk3h+BG7Fu/INFpUIm69lvCywcmVZRH+nIF3
+Nm1aK7u7yC/mmDbxqaZ7Tq+2J+1rJoVTmhkltI55tUfLlvpXJLtYdBsvrU07DbEt
+G8o2PXppLuh9aRI3uRS0jNMCBDo1XQKCAQAa2CsPi/ey9JzgUkBJvVusv7fF8hYB
+4Nno4PXiRezIGbT9WitZU5lQhfw0g9XokyQqKwSS6iEtuSRE92P6XLB0jswQQ/Jc
+5yWX9DqjKKE4dtpS4/VfkdfE6daIqtFCfE3gybnah/FWPAtYY4iC1207lZQjAp91
+OFOV2sfpk4ZIwnSJBvY0O5Brt/nbHkFUzxJRFgERD7zRrFOU9mZdEUfR9jvj4xlI
+NcKeaYuoa4nWwuLEEzNTQqcS8ccOrpGTZQP2ffpyZdY42q4N8UggTdAcwOtQ6a6L
+D3U+YcnG00aa3FnNN5EjOnY4FeIUJwpqzB8mDc0ztHdwOoJhDETWroDq
+-----END RSA PRIVATE KEY-----`
+)
+
+func getPrivateKey(t *testing.T) *rsa.PrivateKey {
+	pemBlock, _ := pem.Decode([]byte(privateKeyPem))
+	privateKey, err := x509.ParsePKCS1PrivateKey(pemBlock.Bytes)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return privateKey
+}
+
+func setupTestService(t *testing.T) *Service {
+	svc := &Service{
+		log:  log.NewNopLogger(),
+		keys: map[string]crypto.Signer{"default": getPrivateKey(t)},
+	}
+	return svc
+}
+
+func TestEmbeddedKeyService_GetJWKS(t *testing.T) {
+	svc := &Service{
+		log: log.NewNopLogger(),
+		keys: map[string]crypto.Signer{
+			"default": getPrivateKey(t),
+			"other":   getPrivateKey(t),
+		},
+	}
+	jwk := svc.GetJWKS()
+
+	require.Equal(t, 2, len(jwk.Keys))
+}
+
+func TestEmbeddedKeyService_GetJWK(t *testing.T) {
+	type args struct {
+		keyID string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    jose.JSONWebKey
+		wantErr bool
+	}{
+		{name: "creates a JSON Web Key successfully",
+			args: args{"default"},
+			want: jose.JSONWebKey{
+				Key: getPrivateKey(t).Public(),
+				Use: "sig",
+			},
+			wantErr: false,
+		},
+		{name: "returns error when the specified key was not found",
+			args:    args{"not-existing-key-id"},
+			want:    jose.JSONWebKey{},
+			wantErr: true,
+		},
+	}
+	svc := setupTestService(t)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := svc.GetJWK(tt.args.keyID)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("EmbeddedKeyService.GetJWK() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("EmbeddedKeyService.GetJWK() = %v, want %v", got, tt.want)
+			}
```

Comment:
Nit. No obligation there but I think we tend to use require in the code base:
```suggestion
			if tt.wantErr {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
			require.Equal(t, got, tt.want)
```